templates/users/login.html

```diff
@@ -1,12 +1,37 @@
 {% extends 'base.html' %}
 
-{% block title %}Login{% endblock %}
+{% block title %}
+  Login
+{% endblock %}
 
 {% block content %}
-<h2>Login</h2>
-<form method="post">
+  <h2>Login</h2>
+
+  <form method="post">
     {% csrf_token %}
-    {{ form.as_p }}
-    <button type="submit">Login</button>
-</form>
+    <fieldset>
+      <div>
+        <label for="id_username" class="form-label mt-4">Email address</label>
+        <input type="text" name="username" maxlength="254" autofocus required class="form-control {% if form.username.errors %}is-invalid{% endif %}" id="id_username" aria-describedby="emailHelp" placeholder="Enter email" />
+        {% for error in form.username.errors %}
+          <div class="invalid-feedback">{{ error }}</div>
+        {% endfor %}
+      </div>
+      <div>
+        <label for="id_password" class="form-label mt-4">Password</label>
+        <input type="password" name="password" class="form-control {% if form.password.errors %}is-invalid{% endif %}" id="id_password" placeholder="Password" autocomplete="off" />
+        {% for error in form.password.errors %}
+          <div class="invalid-feedback">{{ error }}</div>
+        {% endfor %}
+        {% if form.non_field_errors %}
```

Comment:
This is not the correct place to show generic non field error, ideally they need to be on the top. Above the email address container div.